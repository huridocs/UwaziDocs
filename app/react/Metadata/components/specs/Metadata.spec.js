import React from 'react';

import { shallow } from 'enzyme';

import Metadata from '../Metadata';


describe('Metadata', () => {
  let props;

  function testSnapshot() {
    const component = shallow(<Metadata {...props} />);
    expect(component).toMatchSnapshot();
  }

  beforeEach(() => {
    props = {};
  });

  const prepareMultivalueMetadata = () => [
    { name: 'label_array', label: 'label array', value: [{ value: 'first_value', url: 'url1' }, { value: 'second_value', url: 'url2' }] }
  ];

  const testMetadata = (metadata) => {
    props.metadata = metadata;
    testSnapshot();
  };

  it('should not render metadata without value', () => {
    testMetadata([
      { label: 'Label', value: 'string value', name: 'label' },
      { label: 'Label2', name: 'label2' }
    ]);
  });

  it('should render string values', () => {
    testMetadata([{ label: 'Label', value: 'string value', name: 'label' }]);
  });

  it('should render array values separated by ", "', () => {
    testMetadata([{ name: 'label_array', label: 'label array', value: [{ value: 'first_value' }, { value: 'second_value' }] }]);
  });

  it('should render an image field', () => {
    testMetadata([{ name: 'image_label', label: 'Image Label', value: 'http://some.url.com/image.jpg', type: 'image', style: 'cover' }]);
  });

  it('should render a media field', () => {
    testMetadata([{ name: 'media_label', label: 'Media Label', value: 'http://youtube.com/videoid', type: 'media' }]);
  });

  it('should render a Markdown when the metadata is type mardown', () => {
    testMetadata([{ name: 'label_array', label: 'label array', value: 'some markdown text', type: 'markdown' }]);
  });

  it('should render a Geolocation viewer when the metadata is type geolocation', () => {
    testMetadata([{ name: 'geolocation_label', label: 'Geolocation Label', value: [{ lat: 13, lon: 7 }], type: 'geolocation', onlyForCards: true }]);
  });

  it('should render property not have this item when type is null', () => {
    testMetadata([{ name: 'metadata_without_property', label: 'metadata without property', type: null }]);
  });

  it('should render sorted property with sorted styles', () => {
    testMetadata([{ name: 'sortedby', label: 'sortedBy', value: 'string value', sortedBy: true }]);
  });

  it('should render links when the property has url', () => {
    testMetadata([{ name: 'withurl', label: 'withUrl', value: 'string value', url: 'url' }]);
  });

  it('should render links with icons if propery has url and icon', () => {
    testMetadata([{ name: 'withurl', label: 'withUrl', value: 'string value', url: 'url', icon: { _id: 'icon', type: 'Icons' } }]);
  });

  it('should render links when multiple properties have url', () => {
    testMetadata(prepareMultivalueMetadata());
  });

  it('should not render an empty list', () => {
    testMetadata([{ name: 'label_array', label: 'label array', value: [] }]);
  });

  describe('when passing compact prop', () => {
    it('should pass it to ValueList', () => {
      props.metadata = prepareMultivalueMetadata();
      props.compact = true;
      testSnapshot();
    });
  });
});
