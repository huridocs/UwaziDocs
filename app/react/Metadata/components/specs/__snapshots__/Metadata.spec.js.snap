// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metadata should not render an empty list 1`] = `<React.Fragment />`;

exports[`Metadata should not render metadata without value 1`] = `
<React.Fragment>
  <dl
    key="Label"
  >
    <dt>
      Label
    </dt>
    <dd
      className=""
    >
      string value
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata should render a Markdown when the metadata is type mardown 1`] = `
<React.Fragment>
  <dl
    key="label array"
  >
    <dt>
      label array
    </dt>
    <dd
      className=""
    >
      <MarkdownViewer
        html={false}
        lists={Array []}
        markdown="some markdown text"
      />
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata should render array values separated by ", " 1`] = `
<React.Fragment>
  <dl
    key="label array"
  >
    <dt>
      label array
    </dt>
    <dd
      className=""
    >
      <ValueList
        compact={false}
        property={
          Object {
            "label": "label array",
            "value": Array [
              Object {
                "value": "first_value",
              },
              Object {
                "value": "second_value",
              },
            ],
          }
        }
      />
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata should render links when multiple properties have url 1`] = `
<React.Fragment>
  <dl
    key="label array"
  >
    <dt>
      label array
    </dt>
    <dd
      className=""
    >
      <ValueList
        compact={false}
        property={
          Object {
            "label": "label array",
            "value": Array [
              Object {
                "url": "url1",
                "value": "first_value",
              },
              Object {
                "url": "url2",
                "value": "second_value",
              },
            ],
          }
        }
      />
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata should render links when the property has url 1`] = `
<React.Fragment>
  <dl
    key="withUrl"
  >
    <dt>
      withUrl
    </dt>
    <dd
      className=""
    >
      <Connect(I18NLink)
        to="url"
      >
        string value
      </Connect(I18NLink)>
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata should render property not have this item when type is null 1`] = `
<React.Fragment>
  <dl
    key="metadata without property"
  >
    <dt>
      metadata without property
    </dt>
    <dd
      className=""
    >
      No property
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata should render sorted property with sorted styles 1`] = `
<React.Fragment>
  <dl
    key="sortedBy"
  >
    <dt>
      sortedBy
    </dt>
    <dd
      className="item-current-sort"
    >
      string value
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata should render string values 1`] = `
<React.Fragment>
  <dl
    key="Label"
  >
    <dt>
      Label
    </dt>
    <dd
      className=""
    >
      string value
    </dd>
  </dl>
</React.Fragment>
`;

exports[`Metadata when passing compact prop should pass it to ValueList 1`] = `
<React.Fragment>
  <dl
    key="label array"
  >
    <dt>
      label array
    </dt>
    <dd
      className=""
    >
      <ValueList
        compact={true}
        property={
          Object {
            "label": "label array",
            "value": Array [
              Object {
                "url": "url1",
                "value": "first_value",
              },
              Object {
                "url": "url2",
                "value": "second_value",
              },
            ],
          }
        }
      />
    </dd>
  </dl>
</React.Fragment>
`;
